### PR TITLE
Add type hints for several pytorch models (batch-3)

### PR DESCRIPTION
**Pull request by nablabits**
# What does this PR do?
Addresses some of the models in https://github.com/huggingface/transformers/issues/16059:
1. ErnieM
	1. `ErnieMForInformationExtraction`
	2. `ErnieMForMultipleChoice`
	3. `ErnieMForQuestionAnswering`
	4. `ErnieMForSequenceClassification`
	5. `ErnieMForTokenClassification`
	6. `ErnieMModel`
2. `EsmForProteinFolding`
3. `GraphormerModel`
4. `InstructBlipQFormerModel`
5. `LayoutLMForMaskedLM`
6. `LukeForEntitySpanClassification`

## Who can review?

@Rocketknight1, please

### CODE REVIEW

File: src/transformers/models/graphormer/modeling_graphormer.py [hunk 1]
```diff
@@ -816,8 +816,8 @@ def forward(
         out_degree: torch.LongTensor,
         spatial_pos: torch.LongTensor,
         attn_edge_type: torch.LongTensor,
-        perturb=None,
-        masked_tokens=None,
+        perturb: Optional[torch.FloatTensor] = None,
+        masked_tokens: None = None,
```
Comment (nablabits) [inline src/transformers/models/graphormer/modeling_graphormer.py:820]: For `perturb` I took what [microsoft/graphormer](https://github.com/microsoft/Graphormer/blob/b8ef0317f45714ab41d6089b5f8d7fc48ed45242/graphormer/tasks/graph_prediction.py#L314) uses.

`masked_tokens` on the other hand, raises a `NotImplemented` whenever it has something that is not None

File: src/transformers/models/instructblip/modeling_instructblip.py [hunk 2]
```diff
@@ -1124,19 +1124,19 @@ def get_extended_attention_mask(
...
-        past_key_values=None,
-        use_cache=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        return_dict=None,
-    ):
+        input_ids: torch.LongTensor,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        query_embeds: Optional[torch.Tensor] = None,
```
Comment (nablabits) [inline src/transformers/models/instructblip/modeling_instructblip.py:1130]: I was unsure of the `dtype`  (I couldn't find any reference) so I left the generic `torch.Tensor`

**Comment by ArthurZucker** [comment]
cc @Rocketknight1

File: src/transformers/models/graphormer/modeling_graphormer.py (hunk 1 above)
Comment (Rocketknight1) [reply src/transformers/models/graphormer/modeling_graphormer.py:820]: Haa, that's funny. Congrats on our first correct type hint of an explicit `None`!

**Review by Rocketknight1 — APPROVED** [review]
Looks great to me, thank you!

**Pull request merged**